echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/DockerEventMonitor.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pipelinetriggers.monitor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.echo.model.Event;
+import com.netflix.spinnaker.echo.model.Pipeline;
+import com.netflix.spinnaker.echo.model.Trigger;
+import com.netflix.spinnaker.echo.model.trigger.DockerEvent;
+import com.netflix.spinnaker.echo.model.trigger.TriggerEvent;
+import com.netflix.spinnaker.echo.pipelinetriggers.PipelineCache;
+import lombok.NonNull;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import rx.Observable;
+import rx.functions.Action1;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+@Component
+public class DockerEventMonitor extends TriggerMonitor {
+
+  public static final String TRIGGER_TYPE = "docker";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final PipelineCache pipelineCache;
+
+  @Autowired
+  public DockerEventMonitor(@NonNull PipelineCache pipelineCache,
+                @NonNull Action1<Pipeline> subscriber,
+                @NonNull Registry registry) {
+    super(subscriber, registry);
+    this.pipelineCache = pipelineCache;
+  }
+
+  @Override
+  public void processEvent(Event event) {
+    if (!event.getDetails().getType().equalsIgnoreCase(DockerEvent.TYPE)) {
```

Comment:
I would prefer to explicitly throw an exception that has enough detail to do some troubleshooting. I don't think the NPE message would contain details of the event itself.